aea/cli/upgrade.py

```diff
@@ -103,23 +110,42 @@ def upgrade_project(ctx: Context) -> None:  # pylint: disable=unused-argument
     """Perform project upgrade."""
     click.echo("Starting project upgrade...")
 
+    interactive = ctx.config.get("interactive", True)
     old_component_ids = ctx.agent_config.package_dependencies
     item_remover = ItemRemoveHelper(ctx, ignore_non_vendor=True)
     agent_items = item_remover.get_agent_dependencies_with_reverse_dependencies()
     items_to_upgrade = set()
-    upgraders: List[ItemUpgrader] = []
+    upgraders: Dict[PackageId, ItemUpgrader] = {}
     shared_deps: Set[PackageId] = set()
     shared_deps_to_remove = set()
     items_to_upgrade_dependencies = set()
 
     # update aea_version in case current framework version is
     version = Version(aea.__version__)
     if not ctx.agent_config.aea_version_specifiers.contains(version):
+        old_aea_version_specifier = ctx.agent_config.aea_version
         new_aea_version_specifier = compute_specifier_from_version(version)
+        click.echo(
+            f"Upgrading version specifier from {old_aea_version_specifier} to {new_aea_version_specifier}."
+        )
         ctx.agent_config.aea_version = new_aea_version_specifier
         ctx.dump_agent_config()
 
+    eject_helper = InteractiveEjectHelper(ctx, agent_items, interactive=interactive)
+    eject_helper.get_latest_versions()
```

Comment:
we have to get the latest versions of all the packages, this makes the command a bit slow

Reply:
we can refine by filtering non-vendor packages, but not if package is in requirements of another package, as done before (or not?)

Reply:
I am ok with how it is. We don't care if the command is not that fast.